org.lflang/src/org/lflang/generator/c/CCmakeCompiler.java

```diff
@@ -71,68 +74,23 @@ public boolean runCCompiler(String file, boolean noBinary) throws IOException {
         Files.createDirectories(buildPath);
-
-        LFCommand compile = compileCmakeCommand(file, noBinary);
-        if (compile == null) {
-            return false;
-        }
-
-        // Use the user-specified compiler if any
-        if (targetConfig.compiler != null) {
-            // cmakeEnv.remove("CXX");
-            if (targetConfig.compiler.equals("g++") || targetConfig.compiler.equals("CC")) {
-                // Interpret this as the user wanting their .c programs to be treated as
-                // C++ files. We can't just simply use g++ to compile C code. We use a 
-                // specific CMake flag to set the language of all .c files to C++.
-            } else {
-                compile.replaceEnvironmentVariable("CC", targetConfig.compiler);
-            }
-            // cmakeEnv.put("CXX", targetConfig.compiler);
-        }
-        
-        ByteArrayOutputStream stderr = new ByteArrayOutputStream();
-        int cMakeReturnCode = compile.run();
-        
-        if (cMakeReturnCode != 0 && fileConfig.getCompilerMode() != Mode.INTEGRATED) {
-            errorReporter.reportError(targetConfig.compiler+" returns error code "+cMakeReturnCode);
-        }
-        
-        int makeReturnCode = 0;
-
-        if (cMakeReturnCode == 0) {            
-            LFCommand build = buildCmakeCommand(file, noBinary);
-            
-            makeReturnCode = build.run();
-            
-            if (makeReturnCode != 0 && fileConfig.getCompilerMode() != Mode.INTEGRATED) {
-                errorReporter.reportError(targetConfig.compiler+" returns error code "+makeReturnCode);
-            }
-            
-        }
-        
-        // For warnings (vs. errors), the return code is 0.
-        // But we still want to mark the IDE.
-        if (stderr.toString().length() > 0 && fileConfig.getCompilerMode() == Mode.INTEGRATED) {
-            errorReporter.reportError(stderr.toString());
+        int configureReturnCode = configure();
+        int buildReturnCode = 0;
+        // Do not compile if the code generator is being
+        //  invoked from an IDE or text editor without the
+        //  user's explicit request.
+        if (configureReturnCode == 0) {
+            buildReturnCode = build();
         }
-        
-        return ((cMakeReturnCode == 0) && (makeReturnCode == 0));
+        return configureReturnCode == 0 && buildReturnCode == 0;
     }
     
-    
     /**
-     * Return a command to compile the specified C file using CMake.
-     * This produces a C-specific compile command.
-     * 
-     * @param fileToCompile The C filename without the .c extension.
-     * @param noBinary If true, the compiler will create a .o output instead of a binary. 
-     *  If false, the compile command will produce a binary.
+     * Return a command to configure the specified C file
+     * using CMake.
+     * This produces a C-specific configure command.
      */
-    public LFCommand compileCmakeCommand(
```

Comment:
This will also be useful in `org.lflang.federated.FedCLauncher` once we iron out a few issues regarding CMake paths.